lib/user_data_parser/row/row_parser.rb

```diff
@@ -0,0 +1,30 @@
+require_relative "./row.rb"
+
+module UserDataParser
+  class RowParser
+    attr_reader :delimiters, :input_type
+
+    def initialize(delimiters, input_type)
+      @delimiters = delimiters
+      @input_type = input_type
+    end
+
+    def parse(data)
+      values = data.split(delimiters[input_type[:delimiter]])
+      parsed_row_data = {}
+      values.each_with_index do |v, i|
+        parsed_row_data[input_type[:column_order][i]] = v.strip
+      end
+      return Row.new(parsed_row_data)
+    end
+
+    def parse_delimiter_type(data)
+      if data.include?(delimiters[:pipe])
+        return :pipe
+      elsif data.include?(delimiters[:comma])
+        return :comma
+      end
+      :space
```

Comment:
Seems like we're pretty coupled to a specific set of delimiters here. Again, calls out the burden of effort placed on the user in configuring ```input_types```. I'd like to see this program simply reference ```delimiters``` and remain agnostic about what's inside. That could open up the possibility of users specifying insensible delimiters, like a vowel or something, but at that point it may be the user's prerogative.